docs/src/design/range.md

```diff
@@ -157,85 +157,141 @@ Second, we can merge running product columns $p_0$ and $p_1$ into a single colum
+
+- Value of $v$ in the first row is $0$.
+- Value of $v$ in the last row is $65535$.
+
+### 8-bit range checks table
+
+The 8-bit range checks [virtual table](./multiset.md#virtual-tables) is used to enforce the internal correctness of the 16-bit section of the Range Checker (where range checks for user operations and other components are executed). 
+
+This table can be thought of as a virtual table that contains all 8-bit range checks required to ensure correctness of the 16-bit section:
+
+- When a value is range-checked in the 8-bit section of the trace (i.e., the flag indicates that it should be included 1, 2, or 4 times), it is added to the virtual table. The value must be added to the virtual table as many times as that row's flag indicates it should be range-checked.
+- In the 16-bit section of the trace, a single 8-bit range check on $v' - v$ is required at each step. Therefore, a single 8-bit range check of $v' - v$ is removed from the virtual table at each step.
+
+The running product column $p_0$ is used to keep track of the state of the table.
+
+To simplify the notation, we'll first define variable $z$, which represents how a row in the execution trace is reduced to a single value.
 
 $$
 z = (\alpha + v)^4 \cdot f_3 + (\alpha + v)^2 \cdot f_2 + (\alpha + v) \cdot f_1 + f_0
 $$
 
-We'll compute the 16-bit running product in column $p_1$. Transition constraints for this column are fairly straightforward:
+In the 8-bit section ($t = 0$), one, two, or four 8-bit range checks are added to the virtual table at each step and $p_0$ is updated as follows:
 
 $$
-p'_1 = p_1 \cdot (z \cdot t - t + 1)
+p'_0 = p_0 \cdot z
 $$
 
-Thus, when $t = 0$, value in $p_1$ does not change, but when $t = 1$, the next value in $p_1$ is computed by multiplying the current value by $z$.
+Note that if $f_0$ is true then the value of $p_0$ does not change, so no 8-bit range checks are added to the virtual table.
 
-For the 8-bit running product we'll use column $p_0$. Constraints against this column are a bit more complicated. The complication stems from the fact that for the 8-bit section of the table, we want to build up the product, but for the 16-bit section of the table, we want to reduce it. The constraints look as follows:
+In the 16-bit section ($t = 1$), one 8-bit range check is removed from the virtual table at each step and $p_0$ is updated as follows:
 
 $$
-p'_0 \cdot ((\alpha + v' - v) \cdot t - t + 1) = p_0 \cdot (z - z \cdot t + t)
+p'_0 \cdot (\alpha + v' - v) = p_0
 $$
 
-Thus, when $t = 0$ (we are in the 8-bit section), the above expression reduces to:
+The above actually enforces that $p'_0 = p_0 / (\alpha + v' - v)$.
 
-$$
-p'_0 = p_0 \cdot z
+These two updates, which first build up the product and then reduce it, can be combined into a single constraint:
+
+> $$
+p'_0 \cdot ((\alpha + v' - v) \cdot t - t + 1) = p_0 \cdot (z - z \cdot t + t)
```

Comment:
I'll go through and update this file (as minimally as possible). I didn't want to change it too much in this PR, since I was trying to stick to changes required for the multiset docs & wanted to see how you felt about the changes to the final section of this first